src/clj/game/cards/upgrades.clj

```diff
@@ -1407,7 +1407,8 @@
                                  (agenda? target)
                                  (upgrade? target))
                              (= (second (:zone card)) (second (:zone target)))
-                             (first-event? state :corp :corp-install #(= (-> card :zone second) (-> % first :zone second)))))
+                             (first-event? state :corp :corp-install #(and (= (-> card :zone second) (-> % first :zone second))
```

Comment:
This seems like a good place to use `in-server?` instead of manually checking the `:zone`.

Reply:
Feel free to correct me :) I was trying to use in-same-server? but was unable to make it work :)

Reply:
Huh! If it's not working, then something has gone wrong. I'll take a look.